openrtb_ext/imp_adhese.go

```diff
@@ -0,0 +1,203 @@
+package openrtb_ext
+
+import (
+	"encoding/json"
+
+	"github.com/mxmCherry/openrtb"
+)
+
+type ExtImpAdhese struct {
+	Account  string                  `json:"account"`
+	Location string                  `json:"location"`
+	Format   string                  `json:"format"`
+	Keywords []*AdheseKeywordsParams `json:"targets,omitempty"`
+}
+
+type ExtAdhese struct {
+	CreativeId                string `json:"creativeId"`
+	DealId                    string `json:"dealId"`
+	Priority                  string `json:"priority"`
+	OrderProperty             string `json:"orderProperty"`
+	AdFormat                  string `json:"adFormat"`
+	AdType                    string `json:"adType"`
+	AdspaceId                 string `json:"adspaceId"`
+	LibId                     string `json:"libId"`
+	ViewableImpressionCounter string `json:"viewableImpressionCounter"`
+}
+
+type AdheseKeywordsParams struct {
+	Key    string   `json:"key,omitempty"`
+	Values []string `json:"value,omitempty"`
+}
+
+type AdheseOrigin struct {
+	Origin string `json:"origin"`
+}
+
+type AdheseOpenRtbBid struct {
+	Origin            string              `json:"origin"`
+	OriginInstance    string              `json:"originInstance"`
+	OriginData        openrtb.BidResponse `json:"originData"`
+	Ext               string              `json:"ext"`
+	AdType            string              `json:"adType"`
+	SlotName          string              `json:"slotName"`
+	SlotID            string              `json:"slotID,omitempty"`
+	Height            string              `json:"height"`
+	Width             string              `json:"width"`
+	Body              string              `json:"body"`
+	ImpressionCounter string              `json:"impressionCounter"`
+	Extension         Prebid              `json:"extension"`
+}
+
+type AdheseBid struct {
```

Comment:
I'd suggest to remove everything. See what's actually the bare-minimum of fields to make the adapter work. 
Place the strictly required fields on top. Only then go over the list again, and see what would be useful to add and add only that.